### ISSUE TITLE
Error: block not available: latest

### ISSUE THREAD

**Issue by ltfschoen** [a16z/helios#258]
If I run the **call** example with `cargo run -p helios --example call` it outputs the following. Note that I have omitted the Alchemy API key from the output with XXXXX
```
   Compiling execution v0.4.1 (/opt/execution)
   Compiling client v0.4.1 (/opt/client)
   Compiling helios v0.4.1 (/opt)
    Finished dev [unoptimized + debuginfo] target(s) in 21.57s
     Running `target/debug/examples/call`
[2023-08-18T00:57:28Z INFO  call] Execution RPC URL: "https://eth-mainnet.g.alchemy.com/v2/XXXXX"
[2023-08-18T00:57:28Z INFO  call] Consensus RPC URL: https://www.lightclientdata.org
[2023-08-18T00:57:28Z INFO  call] ["MAINNET"] Client built with external checkpoint fallbacks
[2023-08-18T00:57:28Z DEBUG reqwest::connect] starting new connection: https://www.lightclientdata.org/
[2023-08-18T00:57:28Z DEBUG call] Context: call @ 0x8bb9a8baeec177ae55ac410c429cbbbbb9198cac::renderBroker(uint256) <5>
[2023-08-18T00:57:28Z DEBUG reqwest::connect] starting new connection: https://eth-mainnet.g.alchemy.com/
[2023-08-18T00:57:29Z DEBUG reqwest::async_impl::client] redirecting 'https://www.lightclientdata.org/eth/v1/beacon/light_client/bootstrap/0x7beab8f82587b1e9f2079beddebde49c2ed5c0da4ce86ea22de6a6b2dc7aa86b' to 'https://beacon-nd-995-871-887.p2pify.com/c9dce41bab3e120f541e4ffb748efa60/eth/v1/beacon/light_client/bootstrap/0x7beab8f82587b1e9f2079beddebde49c2ed5c0da4ce86ea22de6a6b2dc7aa86b'
[2023-08-18T00:57:29Z DEBUG reqwest::connect] starting new connection: https://beacon-nd-995-871-887.p2pify.com/
[2023-08-18T00:57:29Z INFO  call] [ABIGEN] 0x8bb9a8baeec177ae55ac410c429cbbbbb9198cac::renderBroker(uint256) -> Response Length: 106476
[2023-08-18T00:57:30Z INFO  call] [ABIGEN] 0x8bb9a8baeec177ae55ac410c429cbbbbb9198cac::renderBroker(uint256, uint256) -> Response Length: 64549
[2023-08-18T00:57:30Z DEBUG call] Calling helios client on block: Latest
Error: block not available: latest
```

and also with the **basic** example

```
$ cargo run -p helios --example basic

    Finished dev [unoptimized + debuginfo] target(s) in 0.63s
     Running `target/debug/examples/basic`
[2023-08-18T10:22:25Z INFO  basic] Using untrusted RPC URL [REDACTED]
[2023-08-18T10:22:25Z INFO  basic] Using consensus RPC URL: https://www.lightclientdata.org
[2023-08-18T10:22:25Z INFO  basic] Built client on network "MAINNET" with external checkpoint fallbacks
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: could not fetch bootstrap rpc error on method: bootstrap, message: error sending request for url (https://www.lightclientdata.org/eth/v1/beacon/light_client/bootstrap/0x69e3bb0e44b707636ff2c1d3385b20c21013c95f867c296c7ee84960edcbd103): error trying to connect: dns error: task 9 was cancelled

Location:
    /root/helios/consensus/src/consensus.rs:304:26', /root/helios/consensus/src/consensus.rs:80:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: block not available: latest

Location:
    client/src/node.rs:217:23
```

the errors were experienced when using https://github.com/a16z/helios/pull/262

**Comment by Perelyn-sama** [comment]
I'm getting this error too

**Comment by ncitron** [comment]
Yes this error is due to some big refactoring we are doing on master that has broke a few things. Thanks for pointing this out, as I hadn't realized we broke that.

I'd suggest using the 0.4.1 tagged branch for now, as this should be stable and mostly bug free.

**Comment by ltfschoen** [comment]
i just tried using branch tagged 0.4.1 but gave same `Error: block not available: latest` error

**Comment by ncitron** [comment]
fixed in #285 

As in the examples now, you need to use `wait_synced()` to ensure that helios has finished syncing before fetching data.
